src/main/java/in/projecteka/consentmanager/consent/repository/ConsentArtefactRepository.java

```diff
@@ -73,4 +80,39 @@ private void updateConsentRequest(String consentRequestId,
         }
     }
 
+    public Mono<ConsentArtefactRepresentation> getConsentArtefact(String consentId) {
+        return Mono.create(monoSink -> dbClient.preparedQuery(SELECT_CONSENT_QUERY, Tuple.of(consentId),
+                handler -> {
+                    if (handler.failed()) {
+                        monoSink.error(new RuntimeException("Failed to retrieve CA.", handler.cause()));
+                    } else {
+                        RowSet<Row> results = handler.result();
+                        if (results.iterator().hasNext()) {
+                            Row row = results.iterator().next();
+                            JsonObject artefact = (JsonObject) row.getValue("consent_artefact");
+                            ConsentArtefact consentArtefact = artefact.mapTo(ConsentArtefact.class);
+                            ConsentArtefactRepresentation representation = ConsentArtefactRepresentation
+                                    .builder()
+                                    .status(getConsentStatus(row.getString("status")))
+                                    .consentDetail(consentArtefact)
+                                    .signature(row.getString("signature"))
+                                    .build();
+                            monoSink.success(representation);
+                        } else {
+                            //monoSink.error(new RuntimeException("No such information exists."));
+                            monoSink.success(null);
+                        }
+                    }
+                }));
+    }
+
+    private ConsentStatus getConsentStatus(String status) {
+        return ConsentStatus.valueOf(status);
+    }
+
+    @SneakyThrows
+    private ConsentArtefact convertToConsentDetail(String jsonStr) {
+        ObjectMapper mapper = new ObjectMapper();
+        return mapper.readValue(jsonStr.getBytes(), ConsentArtefact.class);
```

Comment:
passing just a string to readValue doesn't work, is it?

Reply:
removed. not needed anymore